src/components/PDFView/PDFPasswordForm.js

```diff
@@ -0,0 +1,135 @@
+};
+
+class PDFPasswordForm extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            password: '',
+            validationErrorText: '',
+            isEditingInProgress: true,
+        };
+        this.submitPassword = this.submitPassword.bind(this);
+        this.validateOnBlur = this.validateOnBlur.bind(this);
+        this.updatePassword = this.updatePassword.bind(this);
+    }
+
+    submitPassword() {
+        this.validateOnBlur();
+        if (!_.isEmpty(this.state.password)) {
+            this.setState({isEditingInProgress: false});
+            this.props.onSubmit(this.state.password);
+        }
+    }
+
+    updatePassword(password) {
+        if (!_.isEmpty(password)) {
+            this.setState({validationErrorText: ''});
+        }
+        this.setState({password, isEditingInProgress: true});
+    }
+
+    validateOnBlur() {
+        if (!_.isEmpty(this.state.password)) {
+            return;
+        }
+        this.setState({
+            validationErrorText: this.props.translate('attachmentView.passwordRequired'),
+        });
+    }
+
+    render() {
+        // Use container styles appropriate for screen size.
+        const containerStyles = this.props.isSmallScreenWidth
+            ? styles.pdfPasswordForm.narrowScreen
+            : styles.pdfPasswordForm.wideScreen;
+
+        const buttonStyles = [
+            styles.pdfPasswordForm.confirmButton,
+            {paddingBottom: this.props.insets.bottom},
+        ];
+
+        return (
+            <View style={containerStyles}>
+                <Text style={styles.mb4}>
+                    {this.props.translate('attachmentView.pdfPasswordFormLabel')}
+                </Text>
+                <TextInput
+                    label={this.props.translate('common.password')}
+                    autoCompleteType="off"
+                    autoCorrect={false}
+                    textContentType="password"
+                    onInputChange={this.updatePassword}
+                    returnKeyType="done"
+                    onSubmitEditing={this.submitPassword}
+                    errorText={this.state.validationErrorText}
+                    onBlur={this.validateOnBlur}
+                    autoFocus={this.props.shouldAutofocusPasswordField}
+                    secureTextEntry
+                />
+                {!this.state.isEditingInProgress && this.props.isPasswordInvalid && (
+                    <View style={[styles.flexRow, styles.alignItemsCenter, styles.mt3]}>
+                        <Icon src={Expensicons.Exclamation} fill={colors.red} />
+                        <View style={[styles.flexRow, styles.ml2, styles.flexWrap, styles.flex1]}>
+                            <Text style={styles.mutedTextLabel}>
+                                {this.props.translate('attachmentView.passwordIncorrect')}
+                            </Text>
+                        </View>
+                    </View>
+                )}
```

Comment:
This type of error is called server error in our App and this should always be above the submit button. On bigger screens, this error is shown under the input field which it should not. 

If you want to use this type of error then you will have to move it above the button.let's see if FormAlertWithSubmitButton can fit into the picture.

Reply:
I could use `FormAlertWithSubmitButton` but I'll need to change it so that the button type is configurable. Currently it's hardcoded to `success`. May I update `FormAlertWithSubmitButton` so that the button is configurable?

Reply:
Let's wait on this a bit.